### PR TITLE
Move most of mobile contract calls to use ContractKit wrappers

### PR DESCRIPTION
**Pull request by cmcewen**
### Description

This begins moving more of mobile to use contractkit

### CODE REVIEW

File: packages/mobile/.env [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ENVIRONMENT=local
-DEFAULT_TESTNET=alfajoresstaging
+DEFAULT_TESTNET=integration
```
Comment (annakaz) [inline packages/mobile/.env]: Remove from PR?

File: packages/mobile/package.json [hunk 2]
```diff
@@ -174,6 +174,7 @@
     "postinstall-prepare": "^1.0.1",
     "react-devtools": "^3.6.0",
     "react-dom": "16.9.0",
+    "react-native-bundle-visualizer": "^2.1.1",
```
Comment (annakaz) [inline packages/mobile/package.json:174]: Cool you should post a screenshot of your findings w this?

File: packages/mobile/src/tokens/saga.ts [hunk 3]
```diff
@@ -188,3 +185,12 @@ export function tokenTransferFactory({
     }
   }
 }
+
+export async function getCurrencyAddress(currency: CURRENCY_ENUM) {
```
Comment (annakaz) [inline packages/mobile/src/tokens/saga.ts:189]: Could consider caching these addresses?

File: packages/mobile/src/escrow/saga.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
+import { EscrowWrapper } from '@celo/contractkit/lib/wrappers/Escrow'
+import { StableTokenWrapper } from '@celo/contractkit/lib/wrappers/StableTokenWrapper'
```
Comment (annakaz) [inline packages/mobile/src/escrow/saga.ts:2]: Why can't we just import from `@celo/contractkit`?

File: packages/mobile/__mocks__/@celo/contractkit/index.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import BigNumber from 'bignumber.js'
+
+export const newKitFromWeb3 = () => ({
```
Comment (jmrossy) [inline packages/mobile/__mocks__/@celo/contractkit/index.ts:8]: Would be cool if this lived in contract kit itself so other packages could use it. This file could just import and re-export the mock

**Review by annakaz — APPROVED** [review]
LGTM assuming this was tested w local end to end tests? Thanks for migrating all of this over

File: packages/mobile/package.json (hunk 2 above)
Comment (jmrossy) [inline packages/mobile/package.json:174]: Curious what this revealed. Did you mean to leave it in? If so, should we add a yarn script for it?

File: packages/mobile/src/escrow/saga.ts [hunk 6]
```diff
@@ -46,24 +46,31 @@ function* transferStableTokenToEscrow(action: EscrowTransferPaymentAction) {
...
-    const stableToken: StableToken = yield call(getStableTokenContract, web3)
     const account: string = yield call(getConnectedUnlockedAccount)
 
+    const stableToken: StableTokenWrapper = yield call([
+      contractKit.contracts,
+      contractKit.contracts.getStableToken,
+    ])
+    const escrow: EscrowWrapper = yield call([
+      contractKit.contracts,
+      contractKit.contracts.getExchange,
```
Comment (jmrossy) [inline packages/mobile/src/escrow/saga.ts]: getEscrow?

File: packages/mobile/src/escrow/saga.ts [hunk 7]
```diff
@@ -101,7 +108,10 @@ function* withdrawFromEscrow() {
   try {
     Logger.debug(TAG + '@withdrawFromEscrow', 'Withdrawing escrowed payment')
 
-    const escrow: Escrow = yield call(getEscrowContract, web3)
+    const escrow: EscrowWrapper = yield call([
+      contractKit.contracts,
+      contractKit.contracts.getExchange,
```
Comment (jmrossy) [inline packages/mobile/src/escrow/saga.ts]: getEscrow?

File: packages/mobile/src/escrow/saga.ts [hunk 8]
```diff
@@ -46,24 +46,31 @@ function* transferStableTokenToEscrow(action: EscrowTransferPaymentAction) {
...
+    ])
+
     Logger.debug(TAG + '@transferToEscrow', 'Approving escrow transfer')
     const convertedAmount = web3.utils.toWei(amount.toString())
-    const approvalTx = stableToken.methods.approve(escrow.options.address, convertedAmount)
+    const approvalTx = stableToken.approve(escrow.address, convertedAmount)
 
-    yield call(sendTransaction, approvalTx, account, TAG, 'approval')
+    yield call(sendTransaction, approvalTx.txo, account, TAG, 'approval')
```
Comment (jmrossy) [inline packages/mobile/src/escrow/saga.ts:64]: This pattern of extracting out the tx object is what I did in verification too. I haven't thought much about a more elegant way to handle this but there probably is one. No need to fix here, just thinking out loud

File: packages/mobile/src/index.d.ts [hunk 9]
```diff
@@ -41,3 +41,66 @@ declare module '@ungap/url-search-params'
 declare module 'react-native-install-referrer'
 declare module 'react-native-secure-key-store'
 declare module 'qrcode'
+declare module 'web3-eth-abi' {
+  export interface ABIDefinition {
```
Comment (jmrossy) [inline packages/mobile/src/index.d.ts:45]: Nit: could we put this into it's own file and import it here?
Also why do we need this?

File: packages/mobile/src/web3/contracts.ts [hunk 10]
```diff
@@ -1,5 +1,5 @@
 import { newKitFromWeb3 } from '@celo/contractkit'
-import { addLocalAccount as web3utilsAddLocalAccount } from '@celo/walletkit'
+import { addLocalAccount as web3utilsAddLocalAccount } from '@celo/contractkit/src/utils/web3-utils'
```
Comment (jmrossy) [inline packages/mobile/src/web3/contracts.ts]: Should we be importing from `contractkit/dist/` or `lib` instead of `src`?

**Review by jmrossy — COMMENTED** [review]
Nice!! Just a few small comments below

File: packages/mobile/src/escrow/saga.ts (hunk 4 above)
Comment (cmcewen) [reply packages/mobile/src/escrow/saga.ts:2]: It's not exported from the index file

**Pull request merged**